processing/src/main/java/io/druid/segment/ColumnSelectorFactory.java

```diff
@@ -27,6 +27,7 @@
 public interface ColumnSelectorFactory
 {
   public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec);
+  public DimensionSelector makeDictEncodedStringDimensionSelector(DimensionSpec dimensionSpec);
```

Comment:
The "why not introduce a new interface" was a question of why not have something not a `DimensionSelector` that is returned instead.

Given how you've described it, why couldn't the handler implement its own DimensionSelector if it wants to return one of those?  Why does it need to all it on the `ColumnSelectorFactory`?